www/profile/inner.js

```diff
@@ -101,40 +101,57 @@ define([
     var sFrameChan;
 
     var addViewButton = function ($container) {
-        if (APP.readOnly) {
-            return;
-        }
-
         var hash = common.getMetadataMgr().getPrivateData().hashes.viewHash;
         var url = APP.origin + '/profile/#' + hash;
-
-        $('<button>', {
-            'class': 'btn '+VIEW_PROFILE_BUTTON,
-        }).text(Messages.profile_viewMyProfile).click(function () {
-            window.open(url, '_blank');
-        }).appendTo($container);
-
-        $('<button>', {
-            'class': 'btn btn-primary '+VIEW_PROFILE_BUTTON,
-        }).append(h('i.fa.fa-shhare-alt'))
-          .append(h('span', Messages.shareButton))
-          .click(function () {
-            Clipboard.copy(url, (err) => {
-                if (!err) { UI.log(Messages.shareSuccess); }
+        // Only add view and share buttons if not read-only
+        if (!APP.readOnly) {
+            var $blockView = $('<div>', {class: LINK_ID}).appendTo($container);
+            var $button = $('<button>', {
+                'class': 'btn ' + VIEW_PROFILE_BUTTON,
+                'aria-labelledby': 'cp-view-button'
+            }).appendTo($blockView);
+            $button.append(
+                h('span', { id: 'cp-view-button' }, Messages.profile_viewMyProfile)
+            );
+            $button.click(function () {
+                window.open(url, '_blank');
             });
-        }).appendTo($container);
+        }
+        
+        // Always create the description container
+        var $blockDescription = $('<div>', {id: DESCRIPTION_ID}).appendTo($container);
+        APP.$descriptionContainer = $blockDescription;
+        APP.$description = $('<div>', {
+            'id': 'cp-app-profile-description-info'
+        }).appendTo($blockDescription);
+        
+        // Add the share button only for editable profiles
+        if (!APP.readOnly) {
+            var $blockShare = $('<div>', {class: LINK_ID}).appendTo($container);
+            $('<button>', {
+                'class': 'btn btn-primary ' + VIEW_PROFILE_BUTTON, 'aria-labelledby': 'cp-share-button'
+            }).append(
+                h('i', { 'class': 'fa fa-share-alt', 'aria-hidden': 'true'})
+            ).append(
+                h('span', {'id': 'cp-share-button'}, Messages.shareButton)
+            ).click(function () {
+                Clipboard.copy(url, (err) => {
+                    if (!err) { UI.log(Messages.shareSuccess); }
+                });
+            }).appendTo($blockShare);
+        }
     };
-
+    
     var addDisplayName = function ($container) {
-        var $block = $('<div>', {id: DISPLAYNAME_ID}).appendTo($container);
+        var $block = $('<div>', {class: DISPLAYNAME_ID}).appendTo($container);
```

Comment:
I don't understand this change. This makes an element and its child having the same class and applying styles to each other. I see that the class adds margins so each margin is applied twice.